### PR TITLE
[add] Mautic community catalog.

### PR DESCRIPTION
**Pull request by ghost**
This PR adds support for using the Mautic Docker container in the Rancher community catalog.

Mautic is an Open Source Marketing Automation Software.

Stack based on official Mautic version: latest

[mautic/mautic - Docker Hub](https://hub.docker.com/r/mautic/mautic/)

Below follow some screenshots of the Mautic Rancher catalog entry:

## Searching the catalog

rancher-mautic-1

## Creating the stack
	
rancher-mautic-2

### CODE REVIEW

**Comment by DevKyleS** [comment]
Could you specify a specific version of mautic instead of using latest? It's easier to know in Rancher GUI when this catalog entry gets updated instead of having to search manually.
Also have the template version match the above version of mautic instead of saying v1.0.0?

**Comment by KohsakuSaito** [comment]
Thank you for reply.

No, It couldn't specify now. 
but, I think it could specify next "pull request".

You mean config.yml version is equal to rancher-compose.yml version?
And Mautic version too? (ex: 2.11.0)

**Comment by DevKyleS** [comment]
@KohsakuSaito Yes, instead of saying "v1.0.0" use the Mautic version instead (v2.12), and instead of "mautic/mautic" use something like "mautic/mautic:2.12-apache" (or -fpm?).

File: templates/mautic/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+version: '2'
+services:
+  mautic:
+    image: mautic/mautic
```
Comment (rawmind0) [inline templates/mautic/0/docker-compose.yml]: Don't use generic docker image tag, implicit neither explicit, they are not reproducible. Please use concrete docker image version. 
```
-    image: mautic/mautic
+    image: mautic/mautic:2.12-fpm
OR
+    image: mautic/mautic:2.12-apache
```

File: templates/mautic/0/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      MAUTIC_DB_NAME: ${mautic_db_name}
+      MAUTIC_DB_USER: ${mautic_db_user}
+      MAUTIC_DB_PASSWORD: ${mautic_db_password}
+    links:
+      - mauticdb:mysql
+    ports:
+      - ${public_port}:80
+    labels:
+        io.rancher.container.hostname_override: container_name
+        io.rancher.container.pull_image: always
```
Comment (rawmind0) [inline templates/mautic/0/docker-compose.yml]: Once you set an explicit docker image version, you don't need pull image always...
```
-        io.rancher.container.pull_image: always
```

File: templates/mautic/0/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    image: mautic/mautic
+    environment:
+      MAUTIC_DB_HOST: ${mautic_db_host}
+      MAUTIC_DB_NAME: ${mautic_db_name}
+      MAUTIC_DB_USER: ${mautic_db_user}
+      MAUTIC_DB_PASSWORD: ${mautic_db_password}
+    links:
+      - mauticdb:mysql
+    ports:
+      - ${public_port}:80
```
Comment (rawmind0) [inline templates/mautic/0/docker-compose.yml]: Could you please use load balancer to publish exposed port?? 
```
-    ports:
-      - ${public_port}:80
```

File: templates/mautic/0/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        io.rancher.container.hostname_override: container_name
+        io.rancher.container.pull_image: always
+
+  mauticdb:
+    image: mysql:5.7
+    environment:
+      MYSQL_ROOT_PASSWORD: ${mautic_db_password}
+    labels:
+      io.rancher.container.hostname_override: container_name
+      io.rancher.container.pull_image: always
```
Comment (rawmind0) [inline templates/mautic/0/docker-compose.yml]: Add load balancer...
```
+  mautic-lb:
+    image: rancher/lb-service-haproxy:v0.7.9
+    ports:
+      - ${public_port}:80
```

File: templates/mautic/0/rancher-compose.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+.catalog:
+  name: "Mautic"
+  version: "v1.0.0"
```
Comment (rawmind0) [inline templates/mautic/0/rancher-compose.yml]: Better if you could map mautic version with catalog package version.
```
-  version: "v1.0.0"
+  version: "v2.12"
```

File: templates/mautic/config.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+name: Mautic
+description: |
+  Powerful marketing automation.
+version: v1.0.0
```
Comment (rawmind0) [inline templates/mautic/config.yml]: Better if you could map mautic version with catalog package version.
```
-version: "v1.0.0"
+version: "v2.12"
```

File: templates/mautic/VERSION [hunk 7]
```diff
@@ -0,0 +1 @@
+v1.0.0
```
Comment (rawmind0) [inline templates/mautic/VERSION]: This file is not needed, please remove it.

File: templates/mautic/config.yml [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+category: Marketiing Automation
```
Comment (rawmind0) [inline templates/mautic/config.yml]: Please fix typo.
```
-category: Marketiing Automation
+category: Marketing Automation
```

File: templates/mautic/README.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+Stack based on official Mautic version: latest
+
+[mautic/mautic - Docker Hub](https://hub.docker.com/r/mautic/mautic/)
+
+## Documuents:
+
+Read more in the official pages.
+
+- [Documentation](https://mautic.org/docs/en/index.html).
+- [Developer Documentation](https://developer.mautic.org/).
```
Comment (rawmind0) [inline templates/mautic/README.md]: This file is not needed, please remove it.

File: templates/mautic/0/rancher-compose.yml [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  questions:    
+
+    - variable: public_port
+      description: "public port to access the mautic site"
+      label: "PUBLIC_PORT"
+      required: true
+      default: "80"
+      type: "int"
+
+    - variable: mautic_db_host
```
Comment (rawmind0) [inline templates/mautic/0/rancher-compose.yml]: With this package config, mautic_db_host has fixed value `mauticdb`, not variable.

File: templates/mautic/0/rancher-compose.yml [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+      description: "mautic database host"
+      label: "MAUTIC_DB_HOST"
+      required: true
+      default: "mysql"
+      type: "string"
+
+    - variable: mautic_db_name
```
Comment (rawmind0) [inline templates/mautic/0/rancher-compose.yml]: You are not passing `mautic_db_name` to `mauticdb` service as `MYSQL_DATABASE` env variable. Changes here has no effect in the package.

File: templates/mautic/0/rancher-compose.yml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+      description: "mautic database name"
+      label: "MAUTIC_DB_NAME"
+      required: true
+      default: "mautic"
+      type: "string"
+
+    - variable: mautic_db_user
```
Comment (rawmind0) [inline templates/mautic/0/rancher-compose.yml]: You are not passing `mautic_db_user` to `mauticdb` service as `MYSQL_USER` env variable. Changes here has no effect in the package.

File: templates/mautic/0/rancher-compose.yml [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+      description: "mautic user name"
+      label: "MAUTIC_DB_USER"
+      required: true
+      default: "root"
+      type: "string"
+
+    - variable: mautic_db_password
```
Comment (rawmind0) [inline templates/mautic/0/rancher-compose.yml]: You are not passing `mautic_db_password` to `mauticdb` service as `MYSQL_PASSWORD` env variable, just `MYSQL_ROOT_PASSWORD`.

**Review by rawmind0 — CHANGES_REQUESTED** [review]

**Comment by rawmind0** [comment]
Hi @KohsakuSaito , thanks for the PR and contribute.

Please, take a look to the changes requested.

**Review by DevKyleS — APPROVED** [review]

**Comment by KohsakuSaito** [comment]
@rawmind0 Thank you for review.
I will improve based on your feedback.

**Comment by KohsakuSaito** [comment]
@DevKyleS  @rawmind0  
I tried to improve based on feedback.
Would you check of my changes request?

File: templates/mautic/0/rancher-compose.yml [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+      description: "Mautic version/Docker tag used for the Mautic container images."
+      label: "MAUTIC_VERSION"
+      type: enum
+      default: "2.12-apache"
+      options:
+        - 2.12-apache
+        - 2.11-apache
+        - 2.10-apache
+        - 2.9-apache
+        - 2.8-apache
```
Comment (rawmind0) [inline templates/mautic/0/rancher-compose.yml]: Better to maintain deployed version with catalog package. Here may be make sense to be able to select between apache or fpm version. :)

File: templates/mautic/0/docker-compose.yml [hunk 15]
```diff
@@ -0,0 +1,37 @@
+version: '2'
+services:
+
+  mautic-lb:
+    image: rancher/lb-service-haproxy:v0.7.9
+    ports:
+      - 80:80
```
Comment (rawmind0) [inline templates/mautic/0/docker-compose.yml]: What do you think to add a variable to control what port to publish??
```
-      - 80:80
+      - ${publish_port}:${publish_port}
```

File: templates/mautic/0/rancher-compose.yml [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+
+services:
+  mautic-lb:
+    scale: 1
+    lb_config:
+      certs: []
+      port_rules:
+      - protocol: http
+        service: mautic
+        source_port: 80
```
Comment (rawmind0) [inline templates/mautic/0/rancher-compose.yml]: What do you think to add a variable to control what port to publish??
```
-        source_port: 80
+        source_port: ${publish_port}
```

File: templates/mautic/0/rancher-compose.yml [hunk 17]
```diff
@@ -0,0 +1,71 @@
+version: '2'
+
+catalog:
+  name: "Mautic"
+  version: "v2.12"
+  description: "Mautic provides free and open source marketing automation software available to everyone. Free email marketing software, lead management software and more."
+  uuid: Mautic-0
+  minimum_rancher_version: v0.51.0
+  questions:
```
Comment (rawmind0) [inline templates/mautic/0/rancher-compose.yml]: What do you think to add a variable to control what port to publish??
```
    - variable: "publish_port"
      description: "Public port to publish."
      label: "Public port:"
      required: true
      default: 80
      type: "int"
```

**Review by rawmind0 — CHANGES_REQUESTED** [review]

**Comment by rawmind0** [comment]
Hi @KohsakuSaito , thanks for the changes. 

Please, take a look to the changes requested.

File: templates/mautic/0/rancher-compose.yml (hunk 14 above)
Comment (DevKyleS) [reply templates/mautic/0/rancher-compose.yml]: I like this idea unless we stick to fpm and add nginx in front of it for better load balancing/performance

**Comment by DevKyleS** [comment]
@KohsakuSaito Any luck on those changes?

**Comment by exocode** [comment]
Any progress on that PR?

**Comment by cjellick** [comment]
Closing this due to staleness. Feel free to open a new PR if there's still a relevant change to be made. Thanks!

**Pull request closed without merging**